src/Opa.Wasm/OpaPolicy.cs

```diff
@@ -187,12 +200,45 @@ private void ReadAbiVersionGlobals()
 			}
 		}
 
-		public string Evaluate(string json, bool disableFastEvaluate = false)
+		private IOpaResult<T> TypedOutputDeserialize<T>(string evalOutput) where T : notnull
+		{
+			var fullResult = _serializer.Deserialize<OpaResult<T>[]>(evalOutput);
+			var resultObj = fullResult.Length > 0 ? fullResult[0] : null;
+
+			if (null != resultObj)
+			{
+				resultObj.JsonOutput = evalOutput;
+			}
+
+			return resultObj;
+		}
+
+		public IOpaResult<T> Evaluate<T>(object input, bool disableFastEvaluate = false)
+		{
+			// TBD: Passing a string or value type could be unintentional here, should we stop the user from hurting themselves?
+			//if (input.GetType() == typeof(string))
+			//{
+			//	throw new NotSupportedException("The 'string' type argument is not supported.");
+			//}
```

Comment:
If you pass a value object, something still has to happen to get it written to the shared memory buffer for the wasm to evaluate. Not serializing it implies you'll call `opa_parse_value` instead of `opa_parse_json`.  Maybe there's a performance benefit there, but it's definitely simpler to just serialize everything and always load with json.